applications/PfemFluidDynamicsApplication/custom_processes/generate_new_nodes_before_meshing_process.hpp

```diff
@@ -810,10 +824,11 @@ class GenerateNewNodesBeforeMeshingProcess
 		//getting the data of the solution step
 		array_1d<double, 3>& node_data = MasterNode->FastGetSolutionStepValue(variable, step);
 		  
-		array_1d<double, 3>& node0_data = SlaveNode1->FastGetSolutionStepValue(variable, step);
-		array_1d<double, 3>& node1_data = SlaveNode2->FastGetSolutionStepValue(variable, step);
+		array_1d<double, 3> node0_data = SlaveNode1->FastGetSolutionStepValue(variable, step);
+		array_1d<double, 3> node1_data = SlaveNode2->FastGetSolutionStepValue(variable, step);
```

Comment:
I would use 'const array_1d<double, 3>&' or at least '&'. Otherwise, node0_data and node1_data are full vectors allocated temporarily.